apollo-router/src/batching.rs

```diff
@@ -178,15 +175,21 @@ pub(crate) struct BatchQueryInfo {
     sender: oneshot::Sender<Result<SubgraphResponse, BoxError>>,
 }
 
-// #[derive(Debug)]
+#[derive(Debug)]
 pub(crate) struct Batch {
     /// A sender channel to communicate with the batching handler
     senders: Mutex<Vec<Option<mpsc::Sender<BatchHandlerMessage>>>>,
 
     /// The spawned batching handler task handle
     ///
-    /// Note: This _must_ be aborted or else the spawned task may run forever.
-    spawn_handle: JoinHandle<()>,
+    /// Note: We keep this as a failsafe. If the task doesn't terminate _before_ the batch is
+    /// dropped, then we will abort() the task on drop.
+    // spawn_handle: JoinHandle<()>,
+    spawn_handle: JoinHandle<Result<(), BoxError>>,
+
+    /// What is the size (number of input operations) of the batch?
+    #[allow(dead_code)]
```

Comment:
Do we want this to be allowed if nothing is using it?

Reply:
For now. I'm going to do another pass before starting code review and I'm trying to make my mind up if this is useful (for debug logs) or not. I feel like it might be useful to have it appear in the `debug` representation of a `Batch`, but `rustc` can't work that out.